### ISSUE TITLE
How to create enlarged area of existing world-countries map?

### ISSUE THREAD

**Issue by chavisclark** [neveldo/jQuery-Mapael#381]
Hey @neveldo I've been trying to wrap my head around this issue but can't figure out the best way to approach it. 

## **Problem**
The islands, in particular, are super small and I'd like to enlarge one set of islands only.

## **My Attempt**
My initial thought was to simply enlarge the target area of the world-countries svg map in Adobe Illustrator, recreate it using the svg to map tool (https://www.vincentbroute.fr/mapael/svg-to-mapael.php) (svg in[ codepen](https://codepen.io/chavis/pen/xWaEoa)) and use the getCoord() generator (https://www.vincentbroute.fr/mapael/getcoords.php) to associate the new x,y coordinates with the (lat, lon) of my specific area. In my case, Hawaii.

_These are the values I used that configured the function below:_
![captura de pantalla 2018-04-04 a las 1 43 24 p m](https://user-images.githubusercontent.com/17089091/38328203-e1dec6d8-380f-11e8-8081-ba4c09cbbef0.png)

```
                      getCoords : function (lat, lon) {
                        //factors & offsets for normal map
                        var xfactor = 2.752;
                        var xoffset = 473.75;
                        var yfactor = -2.753;
                        var yoffset = 231;
                        // exception for the enlarged area (Hawaii)
                        if (lat < 21.813301 && lon > -154.808557) {
                            xfactor = 2.9406472621931;
                            xoffset = 515.23735930611;
                            yfactor = -12.637933235978;
                            yoffset = 280.67504169429;                            
                        }
                        var x = (lon * xfactor) + xoffset;
                        var y = (lat * yfactor) + yoffset;

                        return {x : x, y : y};
                    },
```

This results in my plots landing all over the place. 

## **My thoughts on why this isn't working**
The problem could be how I calculated the x,y coordinates: https://codepen.io/chavis/pen/xWaEoa

Or how I'm setting up the conditional logic in the getCoords() function.

Is this possible to achieve with this plugin and if so can you provide guidance?

**Comment by chavisclark** [comment]
Sorry to keep opening and closing issues! 🙈 But I figured it out.

The problem was that I needed to fix my **x** and **y** coordinates. The initial map I used was way bigger than the new map I generated. Even though the SVG and proportions were the same (except the enlarged area), when the height and width changed, so did my **x** and **y** and I didn't think to update it. 

So the process I explained in "My Attempt" was correct but: 

1. The x and y I entered in the getCoords generator needed to be adjusted to correspond with the height and width of the svg.

2. I needed to fix and adjust my conditional logic inside getCoords() to only target the specific area I needed to manipulate.

I'll probably make a blog post about it and write a more in-depth look at the steps I took, especially since I use Adobe Illustrator.

Thanks for providing such a great tool!
